### ISSUE TITLE
Logging adapter in readme doesn't resolve log level names

### ISSUE THREAD

**Issue by theunkn0wn1** [Delgan/loguru#154]
## summary
The readme provides a `InterceptHandler` class to hook existing `logging` calls and forward them into loguru. The hook itself works, however does not resolve log levels. 
The result is we get unexpected loguru output, where instead of getting the localized name `'INFO'` we get `'Level 20'`. 

```py
>>> logger.debug("loguru debug")
2019-10-13 21:21:33.728 | DEBUG    | __main__:<module>:1 - loguru debug
>>> logging.debug("logging debug")
2019-10-13 21:21:33.729 | Level 10 | logging:handle:905 - logging debug
>>>
```

This incorrect output also breaks loguru's color formatting
![image](https://user-images.githubusercontent.com/3110986/66722633-10572e80-edc5-11e9-94e1-73186165fb9a.png)

It appears that Loguru does not resolve integer log levels it can understand for message formatting.

## Minimum reproduceable example
```py
import logging
from loguru import logger

# class directly applied from the README
class InterceptHandler(logging.Handler):
    def emit(self, record):
        # Intercepts standard logging messages for the purpose of sending them to loguru
        logger_opt = logger.opt(depth=1, exception=record.exc_info)
        logger_opt.log(record.levelno, record.getMessage())


logging.basicConfig(handlers=[InterceptHandler()], level=0)


logger.debug("loguru debug")
logging.debug("logging debug")

logger.info("loguru info")
logging.info("logging info")


logger.warning("loguru warning")
logging.warning("logging warning")

logger.error("loguru error")
logging.error("logging error")

logger.critical("loguru critical")
logging.critical("logging critical")
```
## Versions
```
>>> loguru.__version__
'0.3.2'
```
- OS: Windows 10 x64, Ubuntu Xenial
- CLI environment: Powershell, bash 

## Potential fix
After debugging and analyzing how loguru handles its own logging calls, it appears that its expecting string level names. e.g. `"DEBUG"` and `"CRITICAL"`. 
With this expectation in mind, this bug can be resolved by having the `InterceptHandler` resolve the `logging` log level name prior to delegating the event to loguru.

It would be a one-line patch
```py
class InterceptHandler(logging.Handler):
    def emit(self, record):
        # Intercepts standard logging messages for the purpose of sending them to loguru
        logger_opt = logger.opt(depth=1, exception=record.exc_info)
        # resolve the log level to a name, then emit the message to Loguru
        logger_opt.log(logging.getLevelName(record.levelno), record.getMessage())
        # logger_opt.log(record.levelno, record.getMessage())
```
![image](https://user-images.githubusercontent.com/3110986/66722777-9b84f400-edc6-11e9-9984-7f91f1ed2a7d.png)

I will open a PR soon to this effect.

**Comment by Delgan** [comment]
Hi @theunkn0wn1, thanks for this detailed bug report! :+1: 

There is one caveat with the `logging.getLevelName()` workaround: it will raise an error while calling `logger.log()` because the level is not registered in `loguru`. For example, try with `logging.log(33, "Boom")`.

**Comment by theunkn0wn1** [comment]
yeah that does appear to cause a KeyError, interesting.

Does logaru expose any mechanism for resolving an integer log level to a logaru `Level` object?
I see there is the internal `logaru._logger._levels: Dict[str: logaru._logger.Level]` which maps between level names and their corresponding levels.

As an alternate solution that would require changes to logaru itself,

We could have a second mapping, `_levels_by_number: Dict[int, logaru._logger.Level]`
on `loguru._logger.Logger._log : 1593` we would check against this dictionary if the `level_id` is None

Worst case we fall back to the existing solution of treating unknown log levels as `Level {no}`

I believe logaru has a mechanism for adding custom log levels, so that would need to be a implementation consideration.

Your thoughts?

**Comment by Delgan** [comment]
> Does logaru expose any mechanism for resolving an integer log level to a logaru Level object?

Nop, and this is on purpose. Levels are identified by their name, not by the severity value. It's perfectly fine to create two levels with different names but the same severity. In such situation, it will not be possible to properly map a severity level to a level name. So, such mapping is plain non-sense in my opinion.

A workaround would be to try to access the level using the [`logger.level()`](https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger.level) method, and fallback to `record.level_no` if the level does not exist:

```python
level_name = record.levelname
try:
    logger.level(level_name)
except ValueError:
    level_name = record.level_no
logger.log(level_name, "Foobar")
```

Unfortunately, this requires more lines of code, making `InterceptHandler()` less elegant. There is others problems anyway with the current snippet, see #78. So, maybe there is no alternative but to expand it.

**Comment by theunkn0wn1** [comment]
Since log levels are bound by name, and multiple levels can have the same severity, then indeed my proposed mapping is nonsense. I had not read enough of the documentation to have noticed that implementation detail.

As you had noted in #78 the two libraries have different internals. I don't really see it to be within the scope of this library to change its internal API to handle `logging` events specifically, they should instead be handled at the adapter. 

The way I see it, the correct place to do any translation from `logging` would be in this Intercept class, as it is the last point before the `logging` event becomes a `loguru` event. While the above try/except snippet makes the adapter less elegant, it appears to be necessary to properly handle compatibility  with standard `logging`. 

![image](https://user-images.githubusercontent.com/3110986/66724354-eeb47200-edd9-11e9-827a-d656c135c42a.png)

**Comment by Delgan** [comment]
Yeah, this is exactly what I think! It seems we reached the same conclusion. It remains to properly define the new implementation.

**Comment by Delgan** [comment]
So, I think I will replace the current snippet:

```python
class InterceptHandler(logging.Handler):
    def emit(self, record):
        # Retrieve context where the logging call occurred, this happens to be in the 6th frame upward
        logger_opt = logger.opt(depth=6, exception=record.exc_info)
        logger_opt.log(record.levelno, record.getMessage())
```

With this one:

```python
class InterceptHandler(logging.Handler):
    def emit(self, record):
        # Get corresponding Loguru level if it exists
        try:
            level = logger.level(record.levelname).name
        except ValueError:
            level = record.levelno

        # Find caller from where originated the logging call
        frame, depth = logging.currentframe(), 2
        while frame.f_code.co_filename == logging.__file__:
            frame = frame.f_back
            depth += 1

        logger.opt(depth=depth, exception=record.exc_info).log(level, record.getMessage())
```

This should fix both #154 (level problem) and #78 (depth problem) in such situation:

```python
import logging
logger_ = logging.getLogger()
logging.info("Foo")
logger_.log(42, "Bar")
```

Obviously, I'm not entirely satisfied with this solution. It's not much pretty. Unfortunately, I don't think there is any other possible workaround, and I do not want to make `InterceptHandler()` / `PropagateHandler()` part of the Loguru API.

There is also the possibility to use the `logger.patch()` to update the Loguru's record dict, but this would not work well with `.enable()` and `.disable()` (because it's resolved before patching) and hence would lead to other problems.

I will probably update the Readme and close this issue in the next few days, if by then I haven't had a revelation. :slightly_smiling_face:

**Comment by Delgan** [comment]
Readme updated. ;)
